explore/src/main/scala/explore/tabs/TargetTabContents.scala

```diff
@@ -265,11 +260,45 @@ object TargetTabContents extends TwoPanels:
       val configuration: Option[BasicConfiguration] = obsConf.map(_._2)
       val wavelength                                = obsConf.map(_._4)
 
-      def setCurrentTarget(programId: Program.Id, oids: ObsIdSet)(
+      def setCurrentTarget(oids: Option[ObsIdSet])(
         tid: Option[Target.Id],
         via: SetRouteVia
       ): Callback =
-        ctx.setPageVia(AppTab.Targets, programId, Focused(oids.some, tid), via)
+        ctx.setPageVia(AppTab.Targets, props.programId, Focused(oids, tid), via)
+
+      def onCloneTarget4Asterism(
+        oldTid:        Target.Id,
+        newTarget:     TargetWithId,
+        obsIdsToClone: ObsIdSet
+      ): Callback =
+        // the new observation ids in the url will be the intersection of what were editing and
+        // what was cloned. This should always have something because otherwise the target editor
+        // would have been readonly.
+        val obsIds4Url = ObsIdSet.fromSortedSet(idsToEdit.idSet.intersect(obsIdsToClone.idSet))
+        // all of the current groups that contain any of the obsIdsToClone
+        val allGroups  = props.programSummaries.model.get.asterismGroups
+          .filterNot(_._1.idSet.intersect(obsIdsToClone.idSet).isEmpty)
+          .map(_._1)
+          .toList
+          .distinct
+        // update the programSummaries
+        props.programSummaries.model.mod { ps =>
+          val newPs = ps.cloneTargetForObservations(oldTid, newTarget, obsIdsToClone)
```

Comment:
I guess this could be inlined

Reply:
Yeah, I think I had more printlns here for debugging purposes.